### ISSUE TITLE
Feature Request: update to typescript-eslint v4

### ISSUE THREAD

**Issue by xiaoxiangmoe** [gund/eslint-plugin-deprecation#19]
see https://github.com/SonarSource/SonarJS/blob/3e3a662019d460bef8729f1e88ae82ff70ecf102/eslint-bridge/package.json#L43-L44

**Comment by gund** [comment]
Hey, thanks for opening an issue!

Do you think if it is enough to just add v4 to this dependency version range?
https://github.com/gund/eslint-plugin-deprecation/blob/3e6f7c3ad5cc8ced45a5867ce7be33a66e8919c1/package.json#L27

**Comment by xiaoxiangmoe** [comment]
```ts
import type { Observable } from "rxjs";

declare const a: Observable<string>;

a.subscribe({
  next: (x) => console.log(x),
});
```

This will cause error in current version of eslint-plugin-deprecation and latest ts+typescript-eslint.

**Comment by xiaoxiangmoe** [comment]
https://github.com/xiaoxiangmoe/eslint-plugin-deprecation-bug

![image](https://user-images.githubusercontent.com/8111351/111452115-2355f700-874d-11eb-9fab-9cc5ba75fa6a.png)
![image](https://user-images.githubusercontent.com/8111351/111452138-294bd800-874d-11eb-98db-bab3f501fb18.png)

As you can see, eslint-plugin-deprecation has different result comparing to [sonarlint-vscode](https://community.sonarsource.com/t/ts-rule-deprecated-apis-should-not-be-used-not-working/40216/3?u=xiaoxiangmoe), which bundle latest sonarjs.

**Comment by gund** [comment]
So with latest `typescript-eslint` this issue is fixed?

**Comment by xiaoxiangmoe** [comment]
It seems that it works in if we use latest typescript-eslint and latest SonarJS.

**Comment by xiaoxiangmoe** [comment]
@gund Can you update it?

**Comment by gund** [comment]
@xiaoxiangmoe can you please confirm if this change is enough?
https://github.com/gund/eslint-plugin-deprecation/pull/20/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R27

**Comment by xiaoxiangmoe** [comment]
@gund No, it is not enough. We should update source codes. see: 
https://github.com/SonarSource/SonarJS/blob/3e3a662019d460bef8729f1e88ae82ff70ecf102/eslint-bridge/src/rules/deprecation.ts

**Comment by gund** [comment]
What exactly is broken?

**Comment by xiaoxiangmoe** [comment]
I don't know. I have tried to pack a new package and use yarn resolution to replace it, but it is not working.

repo: https://github.com/xiaoxiangmoe/eslint-plugin-deprecation-bug

**Comment by gund** [comment]
This is fixed by #39.
